src/DynamicData.Tests/Cache/BindAndSortFixture.cs

```diff
@@ -0,0 +1,372 @@
+using DynamicData.Tests.Domain;
+using FluentAssertions;
+using Xunit;
+
+namespace DynamicData.Tests.Cache;
+
+
+// Bind to a list
+public sealed class BindAndSortToList: BindAndSortFixture
+
+{
+    protected override (ChangeSetAggregator<Person, string> Aggregrator, IList<Person> List) SetUpTests()
+    {
+        var list  = new List<Person>(100);
+        var aggregator = _source.Connect().BindAndSort(list, _comparer).AsAggregator();
+
+        return (aggregator, list);
+    }
+}
+
+// Bind to an observable collection
+public sealed class BindAndSortToObservableCollection : BindAndSortFixture
+
+{
+    protected override (ChangeSetAggregator<Person, string> Aggregrator, IList<Person> List) SetUpTests()
+    {
+        var list = new ObservableCollection<Person>(new List<Person>(100));
+        var aggregator = _source.Connect().BindAndSort(list, _comparer).AsAggregator();
+
+        return (aggregator, list);
+    }
+}
+
+// Bind to a readonly observable collection
+public sealed class BindAndSortToReadOnlyObservableCollection: BindAndSortFixture
+{
+    protected override (ChangeSetAggregator<Person, string> Aggregrator, IList<Person> List) SetUpTests()
+    {
+        var aggregator = _source.Connect().BindAndSort(out var list, _comparer).AsAggregator();
+
+        return (aggregator, list);
+    }
+}
+
+
+
+public abstract class BindAndSortFixture : IDisposable
+{
+
+    private readonly RandomPersonGenerator _generator = new();
+    private readonly ChangeSetAggregator<Person, string> _results;
+    private readonly IList<Person> _boundList;
+
+    protected readonly IComparer<Person> _comparer;
+    protected readonly ISourceCache<Person, string> _source;
+
+
+    public BindAndSortFixture()
+    {
+        _comparer = SortExpressionComparer<Person>.Ascending(p => p.Age).ThenByAscending(p => p.Name);
+        _source = new SourceCache<Person, string>(p => p.Key);
+
+        // It's ok in this case to call VirtualMemberCallInConstructor
+
+#pragma warning disable CA2214
+        // ReSharper disable once VirtualMemberCallInConstructor
+        var args = SetUpTests();
+#pragma warning restore CA2214
+
+        // bind and sort in one hit
+
+        _results = args.Aggregrator;
+        _boundList = args.List;
+
+    }
+
+
+    protected abstract (ChangeSetAggregator<Person, string> Aggregrator, IList<Person> List) SetUpTests();
+
+
+    [Fact]
+    public void AppendAtBeginning()
+    {
+        var people = _generator.Take(100).ToArray();
+        _source.AddOrUpdate(people);
+
+        // check initial data set is sorted
+        _boundList.Count.Should().Be(100);
+        people.OrderBy(p => p, _comparer).SequenceEqual(_boundList).Should().BeTrue();
+
+        //create age 0 to ensure it is inserted first
+        var insert = new Person("_Aaron", 0);
+        _source.AddOrUpdate(insert);
+
+        _boundList.Count.Should().Be(101);
+        var firstItem = _boundList[0];
+
+        firstItem.Should().Be(insert);
```

Comment:
The proper assertion here, in my mind, is the same as above: just assert the bound list is the source items, sorted.

Reply:
I think this is fine to check that a known item has been inserted exactly where we expect it.

With regard to asserting the bound list to be the same as the source.Items sorted, that too would have some merit but I don't think it completely replaces the need for an exact assertion.

Reply:
Technically, if you're asserting that the whole list exactly matches the desired sorting, that *should* mean that asserting a single item's position is superfluous, but I'm also a big fan of clarified intent, and I'm not gonna knock a tiny inefficiency in testing.

Reply:
I like to combine both.  The tests should also not force a cognitive load for the person running or debugging them. Where possible a declarative assertion tells fresh eyes / a newbie exactly what is happening and not just to assert correctness.